src/parser/priest/discipline/modules/spells/Atonement.tsx

```diff
@@ -73,37 +83,41 @@ class Atonement extends Analyzer {
     this.currentAtonementTargets = this.currentAtonementTargets.filter(id => id.target !== atonement.target);
     this.currentAtonementTargets.push(atonement);
     this.totalAtones += 1;
+    // @ts-ignore
     debug && console.log(`%c${this.combatants.players[atonement.target].name} gained an atonement`, 'color:green', this.currentAtonementTargets);
     this.eventEmitter.fabricateEvent({
       type: 'atonement_applied',
+      // @ts-ignore
       timestamp: event.timestamp,
       sourceID: event.sourceID,
       targetID: event.targetID,
     }, event);
   }
 
-  on_byPlayer_refreshbuff(event) {
+  on_byPlayer_refreshbuff(event: RefreshBuffEvent) {
     const spellId = event.ability.guid;
     if (spellId !== SPELLS.ATONEMENT_BUFF.id) {
       return;
     }
 
     // Check if Atonement was refreshed too early
-    let refreshedTarget = this.currentAtonementTargets.find(id => id.target === event.targetID);
+    let refreshedTarget: AtonementTarget | undefined = this.currentAtonementTargets.find(id => id.target === event.targetID);
     if (!refreshedTarget) {
       refreshedTarget = {
         target: event.targetID,
         lastAtonementAppliedTimestamp: this.owner.fight.start_time,
+        atonementExpirationTimestamp: 0,
       };
       debug && console.warn('Atonement: was applied prior to combat');
     }
     const timeSinceApplication = event.timestamp - refreshedTarget.lastAtonementAppliedTimestamp;
     if (timeSinceApplication < ((this.atonementDuration) - IMPROPER_REFRESH_TIME)) {
       this.improperAtonementRefreshes.push(refreshedTarget);
+      // @ts-ignore
       debug && console.log(`%c${this.combatants.players[event.targetID].name} refreshed an atonement too early %c${timeSinceApplication}`, 'color:red', this.currentAtonementTargets);
       this.eventEmitter.fabricateEvent({
         type: 'atonement_refresh_improper',
-        timestamp: event.timestamp,
```

Comment:
I believe these (the timestamps) should stay, and the TS ignore can be moved up. I didn't realize how badly the shape was for the EventEmitter, but once that's fixed the ignores shouldn't be needed